mantidproject/mantid
# Add embedded find/replace dialog to editor widget

This PR is waiting on ~~#24700~~ to reuse some of the memory leaking common functionality.

**Description of work.**
Adds embedded find/replace dialog into the editor widget. To trigger press `Ctrl+F`.

Also fixes the editor widgets (and children) being leaked.



**To test:**
- Press CTRL+F in the editor. Do a bunch of search/replaces. Try to check if all the options work correctly. Try some unicode symbols: `áéóúí`.
- Test that search is performed independently for each editor
- Test pressing escape to close the find/replace, only works if Find/Replace has focus (seems to be `QsciScintilla` limitation - it doesn't allow `Escape` trigger through the editor itself, I am not sure why, it doesn't seem to be used for anything)
- Added `Enter` to search next, `Shift+Enter` to search previous, only works if Find/Replace has focus



Fixes #24531 





---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
